presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerdeFactory.java

```diff
@@ -25,19 +27,42 @@
 {
     private final BlockEncodingSerde blockEncodingSerde;
     private final boolean compressionEnabled;
+    private final boolean spillEncryptionEnabled;
 
-    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled)
+    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled, boolean spillEncryptionEnabled)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.compressionEnabled = compressionEnabled;
+        this.spillEncryptionEnabled = spillEncryptionEnabled;
     }
 
     public PagesSerde createPagesSerde()
     {
+        return createPagesSerdeInternal(Optional.empty());
+    }
+
+    /**
+     * Creates a {@link PagesSerde} instance for use in spilling to disk, which will be encrypted if {@link PagesSerdeFactory#spillEncryptionEnabled}
```

Comment:
Creates a {@link PagesSerde} instance for use in spilling to disk. The data will be encrypted if {@link PagesSerdeFactory#spillEncryptionEnabled} is set.